crates/iroha_cli/src/main.rs

```diff
@@ -1197,6 +1200,232 @@ mod json {
+                return Err(eyre!("signatories and weights must be equal in length"));
+            }
+            let registry_id: TriggerId = format!("multisig_accounts_{}", account.domain())
+                .parse()
+                .unwrap();
+            let account = account.signatory.clone();
+            let signatories = signatories.into_iter().zip(weights).collect();
+            let args = MultisigAccountArgs {
+                account,
+                signatories,
+                quorum,
+                transaction_ttl_secs,
+            };
+            let register_multisig_account =
+                iroha::data_model::isi::ExecuteTrigger::new(registry_id).with_args(&args);
+
+            submit([register_multisig_account], Metadata::default(), context)
+                .wrap_err("Failed to register multisig account")
+        }
+    }
+
+    /// Args to propose a multisig transaction
+    #[derive(Debug, clap::Args)]
+    pub struct Propose {
+        /// Multisig authority of the multisig transaction
+        #[arg(short, long)]
+        pub account: AccountId,
+    }
+
+    impl RunArgs for Propose {
+        fn run(self, context: &mut dyn RunContext) -> Result<()> {
+            let Self { account } = self;
+            let registry_id: TriggerId = format!(
+                "multisig_transactions_{}_{}",
+                account.signatory(),
+                account.domain()
+            )
+            .parse()
+            .unwrap();
+            let instructions: Vec<InstructionBox> = {
+                let mut reader = BufReader::new(stdin());
+                let mut raw_content = Vec::new();
+                reader.read_to_end(&mut raw_content)?;
+                let string_content = String::from_utf8(raw_content)?;
+                json5::from_str(&string_content)?
+            };
+            let instructions_hash = HashOf::new(&instructions);
+            println!("{instructions_hash}");
+            let args = MultisigTransactionArgs::Propose(instructions);
+            let propose_multisig_transaction =
+                iroha::data_model::isi::ExecuteTrigger::new(registry_id).with_args(&args);
+
+            submit([propose_multisig_transaction], Metadata::default(), context)
+                .wrap_err("Failed to propose transaction")
+        }
+    }
+
+    /// Args to approve a multisig transaction
+    #[derive(Debug, clap::Args)]
+    pub struct Approve {
+        /// Multisig authority of the multisig transaction
+        #[arg(short, long)]
+        pub account: AccountId,
+        /// Instructions to approve
+        #[arg(short, long)]
+        pub instructions_hash: iroha::crypto::Hash,
+    }
+
+    impl RunArgs for Approve {
+        fn run(self, context: &mut dyn RunContext) -> Result<()> {
+            let Self {
+                account,
+                instructions_hash,
+            } = self;
+            let registry_id: TriggerId = format!(
+                "multisig_transactions_{}_{}",
```

Comment:
> Isn't the scoping feasible by permission of the trigger authority?

but it's not real scoping because we can't have 2 triggers of the same name. For instance in your case if we could have `multisig_transactions` scoped under `account` you could have multiple instances of the same trigger under different accounts. Instead you have to rely on parsing trigger id

Reply:
I've seen this pattern many times. I think this is something to be considered outside the scope of this PR

Reply:
opened #5160